jupyter_elfinder/elfinder.py

```diff
@@ -453,160 +497,161 @@ def __open(self) -> None:
+                self._response[R_ERROR] = "Access denied"
             else:
                 path = target
         self.__content(path, False)
 
     def __rename(self) -> None:
         """Rename file or dir."""
-        name = self._request.get("name")
-        current = self._request.get("current")
-        target = self._request.get("target")
+        name = self._request.get(API_NAME)
+        current = self._request.get(API_CURRENT)
+        target = self._request.get(API_TARGET)
 
         if not (name and current and target):
-            self._response["error"] = "Invalid parameters"
+            self._response[R_ERROR] = "Invalid parameters"
             return
 
         cur_dir = self.__find_dir(current, None)
 
         if not cur_dir:
-            self._response["error"] = "File not found"
+            self._response[R_ERROR] = "File not found"
             return
 
         cur_name = self.__find(target, cur_dir)
 
         if not cur_name:
-            self._response["error"] = "File not found"
+            self._response[R_ERROR] = "File not found"
             return
         if not self.__is_allowed(cur_dir, "write") and self.__is_allowed(
             cur_name, "rm"
         ):
-            self._response["error"] = "Access denied"
+            self._response[R_ERROR] = "Access denied"
             return
 
         name = self.__check_utf8(name)
 
         if not name or not _check_name(name):
-            self._response["error"] = "Invalid name"
+            self._response[R_ERROR] = "Invalid name"
             return
 
         new_name = os.path.join(cur_dir, name)
 
         if os.path.exists(new_name):
-            self._response["error"] = (
+            self._response[R_ERROR] = (
                 "File or folder with the same name" + "already exists"
             )
             return
 
         self.__rm_tmb(cur_name)
         try:
             os.rename(cur_name, new_name)
-            self._response["added"] = [self.__info(new_name)]
-            self._response["removed"] = [target]
+            self._response[R_ADDED] = [self.__info(new_name)]
+            self._response[R_REMOVED] = [target]
         except OSError:
-            self._response["error"] = "Unable to rename file"
+            self._response[R_ERROR] = "Unable to rename file"
 
     def __mkdir(self) -> None:
         """Create new directory."""
         path = None
         new_dir = None
-        if "name" in self._request and "target" in self._request:
-            name = self._request["name"]
+        if API_NAME in self._request and API_TARGET in self._request:
+            name = self._request[API_NAME]
             name = self.__check_utf8(name)
-            target = self._request["target"]
+            target = self._request[API_TARGET]
             path = self.__find_dir(target, None)
+        # FIXME: The api spec says that it should be `dirs[]`.  # pylint: disable=fixme
         if "dirs" in self._request:
```

Comment:
I think this is wrong. See https://github.com/Studio-42/elFinder/wiki/Client-Server-API-2.1#mkdir

Reply:
You are right, it should be `dirs[]`.